pgmpy/inference/dbn_inference.py

```diff
@@ -0,0 +1,207 @@
+import itertools
+from collections import defaultdict
+
+from pgmpy.models import BayesianModel, DynamicBayesianNetwork
+from pgmpy.factors import Factor
+from pgmpy.factors.Factor import factor_product
+from pgmpy.inference import Inference, BeliefPropagation
+
+
+class DBNInference(Inference):
+
+    def __init__(self, model):
+        """
+        Class for performing inference using Belief Propagation method
+        for the input Dynamic Bayesian Network.
+        Parameters
```

Comment:
@palashahuja Please check the docstrings written in other methods and follow the same format. Also add doctests.